src/main/webapp/script.js

```diff
@@ -316,24 +324,12 @@ class AutocompleteDirectionsHandler {
                 let legEndlat = leg["end_location"].lat();
```

Comment:
Checking for the last iteration of the loop every loop is inefficient; can you think of a tidy way to add this final endpoint AQI value after the loop?